### PR TITLE
finished basic read functionality

### PR DESCRIPTION
**Pull request by Drewbrad4**
Added the MyLibrary index, the borrow index, the borrow show, the borrowed show, and the lended show pages.

### CODE REVIEW

File: app/javascript/components/App.js [hunk 1]
```diff
@@ -20,27 +20,30 @@ import NotFound from './Pages/NotFound'
...
-  render () {
+  render() {
     const {
       logged_in,
       sign_in_route,
       sign_up_route,
       sign_out_route,
       current_user
     } = this.props
+    console.log(current_user);
```
Comment (sturdynut) [inline app/javascript/components/App.js:42]: 🔥 I believe this can be removed

File: app/javascript/components/App.js [hunk 2]
```diff
@@ -49,16 +52,75 @@ export default class App extends React.Component {
...
+                />
+              )
+            }}
+          />
+
+          <Route
+            path="/borrow/:id"
+            render={(props) => {
+              let id = props.match.params.id
+              let book = this.state.books.find(book => book.id === parseInt(id))
```
Comment (sturdynut) [inline app/javascript/components/App.js:85]: If you are doing just a simple string to int conversion, you can also use `Number(id)`.  Up to you!

File: app/javascript/components/Pages/Borrow.js [hunk 3]
```diff
@@ -1,11 +1,34 @@
...
-    }
-}
\ No newline at end of file
+const Borrow = ({ books }) => (
+
+    <>
+        <h3 className="heading">All Books</h3>
+        { books.map((book, index) => {
+            return (
+                <div key={index} className="bookCard">
```
Comment (sturdynut) [inline app/javascript/components/Pages/Borrow.js:11]: This totally works, but as apps grow, it is easy to start having duplicate keys.  React uses these keys to identify the component and to optimize re-renders when [reconciling](https://reactjs.org/docs/faq-internals.html).  So, in this case I would recommend something like: ``key={`book-${index}`}``.

**Review by sturdynut — APPROVED** [review]
🥳  Looks awesome! 🤩
I left some things to consider, but otherwise nice work!

File: app/javascript/components/App.js (hunk 1 above)
Comment (Drewbrad4) [reply app/javascript/components/App.js:42]: Yes, thank you!

File: app/javascript/components/Pages/Borrow.js (hunk 3 above)
Comment (Drewbrad4) [reply app/javascript/components/Pages/Borrow.js:11]: Very interesting! We will definitely edit this stuff and push back up tomorrow. Thank you!

**Comment by talk2MeGooseman** [comment]
Remember your propTypes :D

File: app/javascript/components/Pages/MyLibrary.js [hunk 4]
```diff
@@ -8,7 +9,7 @@ const MyLibrary = ({ books }) => (
         <h2>My Library</h2>
         { books.map((book, index) => {
             return (
-                <div key={index} className="bookCard">
+                <div key={`library-${index}`} className="bookCard">
```
Comment (Drewbrad4) [inline app/javascript/components/Pages/MyLibrary.js:12]: I did have a slight question here. We figured this key obviously needed to be different from our key in Borrow.js. Is this naming convention okay or is there something better for this? The My Library page is an index of all the books owned by the user.

Comment (talk2MeGooseman) [reply app/javascript/components/Pages/MyLibrary.js:12]: Keys are exclusive to within the component, you dont have to worry about how you define keys in some other component outside of this. React is smart enough to namespace them properly. React will get angry at you if your improperly set keys.

**Pull request merged**